### PR TITLE
RFC-413: Feedback from review

### PR DESCRIPTION
**Pull request by timj**
* First draft of use case comparing old with new

### CODE REVIEW

**Comment by timj** [comment]
@gpdf or @TallJimbo I was actioned to come up with a new use case for comparison of pre-release data with a previous data release. I have a first draft here, would you please take a look. I assume we are talking mostly about doing catalog comparisons rather than image comparisons?

We already have a requirement, DMS-REQ-0339, for something close to this, involving a subset of a data release.

**Review by TallJimbo — APPROVED** [review]
One request for clarification, but otherwise looks fine.  It's vague, but I don't think going into detail would necessarily be a worthwhile endeavor here.

File: LDM-592.tex
Comment (TallJimbo) [inline LDM-592.tex]: Does "combined catalogs" just mean "catalogs from both releases"?

**Review by TallJimbo — APPROVED** [review]

File: LDM-592.tex
Comment (timj) [reply LDM-592.tex]: I think I was probably thinking some sort of cross-matched catalog from both releases but "using catalog information from both releases" probably can cover both the "catalogs from both releases" and "cross matched catalog derived from both releases".

**Comment by timj** [comment]
@RobertLuptonTheGood can you take a look at the new re-run use case please? I've tried to take what you said on JIRA and what @TallJimbo has told me and turn it into a use case. @gpdf a reminder that I am waiting for you to sign off on your new use case.

File: LDM-592.tex
Comment (RobertLuptonTheGood) [inline LDM-592.tex]: I'd like to see the system use reruns too, so any processing gets a "rerun" tag (e.g. 2025/DRP/RC2).  Then I can look at those outputs or reprocess part of that data in some different way.

I.e. Everyone will say `butler = Butler(URI, rerun="2025/DRP/RC2")` in their notebooks.  Anyone wanting to reprocess data will say something like `--rerun 2025/DRP/RC2:rhl/tmp` and then `butler = Butler(URI, rerun="rhl/tmp")` to look at the results.

Reruns *supersede* explicit specification of inputs and outputs (except possibly deep in processing scripts that no-one outside NCSA ever sees)

File: LDM-592.tex
Comment (RobertLuptonTheGood) [inline LDM-592.tex:979]: I'm not sure that this is a different case.  If I decide that I want to do more processing I use `--rerun rhl/tmp` to write more outputs, and possibly classes of outputs, to rerun `rhl/tmp`.  Of course, `--rerun rhl/tmp:rhl/foo` (with a different configuration of options/versions) will work too if I want to compare the outputs in `rhl/tmp` to those in `rhl/foo`.

File: LDM-592.tex
Comment (gpdf) [reply LDM-592.tex]: To the extent that this use case came from me, the intent was just that both catalogs would be available through the same Butler in a single job, not to imply that any cross-matching would already have been done.  It was meant to _facilitate_ cross-matching.

File: LDM-592.tex
Comment (gpdf) [inline LDM-592.tex]: I'm happy with this adjustment; it's consistent with the spirit of what I had in mind.

**Comment by gpdf** [comment]
I'm OK with DRP-31 as it stands.  I think in practice it applies to the situation of comparing _any_ two processings of the same data to each other, not just to the situation when one of the two is an actual "Data Release", though.

**Review by gpdf — APPROVED** [review]
(Copied summary comment here, where it should have gone in the first place.)
I'm OK with DRP-31 as it stands. I think in practice it applies to the situation of comparing any two processings of the same data to each other, not just to the situation when one of the two is an actual "Data Release", though.
The other changes in the PR are good.

File: LDM-592.tex
Comment (TallJimbo) [reply LDM-592.tex]: I think the difference between what's in the text and in @RobertLuptonTheGood's comment is policy; it's whether we declare that this is the *only* way processing runs are labeled.

I think that's reasonable (and it will actually be fairly strongly baked into the Gen3 design), but I'll defer to @timj as to whether it belongs in a use case.

File: LDM-592.tex
Comment (TallJimbo) [reply LDM-592.tex:979]: Right; I think as it's currently formatted it's not actually a different use case.  I just wanted to make sure the use case captured the idea that the rerun implicitly contains the inputs (recursively), not just the outputs.

Note, however, that passing `--rerun rhl/tmp:rhl/foo` to a pipe_base argument parser does not compare outputs; it creates a new rerun `rhl/foo` for output that contains (via the parent mechanism, for now) everything in `rhl/tmp`.  If you want to do a comparison that involves looking at the same data ID in different repos you need two `Butlers` (and this is not planned to change with Gen3).

**Comment by RobertLuptonTheGood** [comment]
Fair enough, but I didn't think that Tim's wording captured the intent of my use case.

**Comment by timj** [comment]
@RobertLuptonTheGood Can you provide a clarification to the use case using use-case style language? ie "Some entity does something and then does something else".

File: LDM-592.tex
Comment (RobertLuptonTheGood) [inline LDM-592.tex]: After much discussion with Tim, I think that my problem (as regards use cases) stems from this sentence.  If it said "A Developer runs some processing starting from outputs produced by the Production Operators or another Developer" I'd be happy enough.

Separately, we should modify the developer guide to say that the "rerun" mechanism should be used instead of explicitly specifying input and output directories.

**Comment by timj** [comment]
RFC-413 was implemented.

**Pull request closed without merging**
